xdsl/parser.py

```diff
@@ -1464,6 +1464,25 @@ def try_parse_builtin_arr_attr(self) -> ArrayAttr | None:
     def parse_optional_attr_dict(self) -> dict[str, Attribute]:
         raise NotImplementedError()
 
+    def parse_optional_dictionary_attr_dict(self) -> dict[str, Attribute]:
+        if not self.tokenizer.starts_with("{"):
+            return dict()
+
+        self.parse_characters(
+            "{",
+            "MLIR Attribute dictionary must be enclosed in curly brackets")
```

Comment:
This error message is out of sync now, right? This does not just apply to the MLIR one, but to general ones.

Reply:
Yep